tests/test_new_eff.py

```diff
@@ -152,3 +151,141 @@ def foo( N : size, x : R[N,N] ):
 # for its body.  This can probably distinguish whether certain
 # rewrites are allowed or not.
 
+
+def test_delete_config_basic():
+    @config
+    class CFG:
+        a : index
+        b : size
+
+    @proc
+    def foo( N : size, x : R[N] ):
+        CFG.a = 3
+        for i in seq(0,N):
+            x[i] = x[i] + 1.0
+
+    foo = foo.delete_config('CFG.a = _')
+    print(foo)
+
+
+def test_delete_config_subproc_basic():
+    @config
+    class CFG:
+        a : index
+        b : size
+
+    @proc
+    def do_config():
+        CFG.a = 3
+        CFG.b = 5
+
+    @proc
+    def foo( N : size, x : R[N] ):
+        do_config()
+        for i in seq(0,N):
+            x[i] = x[i] + 1.0
+
+    foo = foo.delete_config('do_config()')
+    print(foo)
+
+def test_delete_config_fail():
+    @config
+    class CFG:
+        a : index
+        b : size
+
+    @proc
+    def foo( N : size, x : R[N] ):
+        CFG.a = 3
+        for i in seq(0,N):
+            if i < CFG.a:
+                x[i] = x[i] + 1.0
+
+    with pytest.raises(SchedulingError,
+                       match='Cannot change configuration value of CFG_a'):
+        foo = foo.delete_config('CFG.a = _')
+        print(foo)
+    
+def test_delete_config_subproc_fail():
+    @config
+    class CFG:
+        a : index
+        b : size
+
+    @proc
+    def do_config():
+        CFG.a = 3
+        CFG.b = 5
+
+    @proc
+    def foo( N : size, x : R[N] ):
+        do_config()
+        for i in seq(0,N):
+            if i < CFG.a:
+                x[i] = x[i] + 1.0
+
+    with pytest.raises(SchedulingError,
+                       match='Cannot change configuration value of CFG_a'):
+        foo = foo.delete_config('do_config()')
+        print(foo)
+
+
+def test_delete_config_bc_shadow():
+    @config
+    class CFG:
+        a : index
+        b : size
+
+    @proc
+    def foo( N : size, x : R[N] ):
+        CFG.a = 34
+        CFG.a = 3
+        for i in seq(0,N):
+            if i < CFG.a:
+                x[i] = x[i] + 1.0
+
+    foo = foo.delete_config('CFG.a = _ #0')
+    print(foo)
+
+
+def test_delete_config_bc_redundant():
+    @config
+    class CFG:
+        a : index
+        b : size
+
+    @proc
+    def foo( N : size, x : R[N] ):
+        CFG.a = 3
+        CFG.a = 3
+        for i in seq(0,N):
+            if i < CFG.a:
+                x[i] = x[i] + 1.0
+
+    foo = foo.delete_config('CFG.a = _ #1')
+    print(foo)
+
+def test_delete_config_fail_bc_not_redundant():
+    @config
+    class CFG:
+        a : index
+        b : size
+
+    @proc
+    def foo( N : size, x : R[N] ):
+        CFG.a = 34
+        CFG.a = 3
+        for i in seq(0,N):
+            if i < CFG.a:
+                x[i] = x[i] + 1.0
+
+    with pytest.raises(SchedulingError,
+                       match='Cannot change configuration value of CFG_a'):
+        foo = foo.delete_config('CFG.a = _ #1')
+        print(foo)
+
+
+
+
+
+
```

Comment:
Don't check in pointless newlines.

Reply:
I do a lot of navigation through files by scrolling, so usually gratuitous newlines are equivalent to a page break of sorts

Reply:
That's a problem with your editor, then... this doesn't belong in the source.

Reply:
IIRC you use Sublime Text? https://stackoverflow.com/a/44185534/2137996

Reply:
This isn't a code editor thing; and I didn't realize it was at the end of the file; it's fine to trim these lines off the end.  Sorry for this going on so long.